src/basilisp/core.lpy

```diff
@@ -4937,7 +4937,8 @@
   ;; old python global ns where the import would have been evaluated
   ;; otherwise.
   (let [form `(import* ~@modules)]
-    `(eval '~form  *ns*)))
+    `(do (eval '~form  *ns*)
+         nil)))
```

Comment:
I was able to get things working with this:

```
(let [form `(do (import* ~@modules)
                  nil)]
    `(eval '~form  *ns*))
```

I'm not clear as to why it fails with the current suggestion.

Reply:
Thanks for the hint, I've posted some analysis with the generated python between success and failure examples in the comments below